t/plugin/prometheus.t

```diff
@@ -632,3 +632,53 @@ GET /apisix/prometheus/metrics
 qr/apisix_/
 --- response_body_unlike eval
 qr/etcd/
+
+
+
+=== TEST 42: fetch the prometheus one shared dict data storage with string
+--- yaml_config
+nginx_config:
+  http:
+    lua_shared_dict:
+      internal-status: 10m
+--- request
+GET /apisix/prometheus/metrics
+--- response_body_like eval
+qr/.*apisix_shared_dict_capacity_bytes{name="internal-status"} \d+(?:.|\n)*
+apisix_shared_dict_free_space_bytes{name="internal-status"} \d+.*/
+
+
+
+=== TEST 43: fetch the prometheus multiple shared dict data
+--- yaml_config
+nginx_config:
+  http:
+    lua_shared_dict:
+      worker-events: 10m
+      upstream-healthcheck: 10m
+      internal-status: 10m
+--- request eval
+["GET /apisix/prometheus/metrics", "GET /apisix/prometheus/metrics", "GET /apisix/prometheus/metrics"]
+--- response_body_like eval
+[qr/.*apisix_shared_dict_capacity_bytes{name="worker-events"} \d+(?:.|\n)*
+apisix_shared_dict_free_space_bytes{name="worker-events"} \d+.*/,
+qr/.*apisix_shared_dict_capacity_bytes{name="upstream-healthcheck"} \d+(?:.|\n)*
+apisix_shared_dict_free_space_bytes{name="upstream-healthcheck"} \d+.*/,
+qr/.*apisix_shared_dict_capacity_bytes{name="internal-status"} \d+(?:.|\n)*
+apisix_shared_dict_free_space_bytes{name="internal-status"} \d+.*/]
+
+
+
+=== TEST 44: fetch the prometheus multiple shared dict data contain not exist shared dict
+--- yaml_config
```

Comment:
Looks like we don't need this test case anymore?

Reply:
I consider it as an example to show that `ngx.shared` is determined not only by the `yaml` file but also by the `ngx_tpl.lua` file.

Reply:
1. The final result of `ngx_tpl.lua` depends on the configuration of `yaml`, I don't understand what you mean, and your test cases are written like this:
````
--- yaml_config
nginx_config:
   http:
     lua_shared_dict:
````
2. You better indicate your test intent in the title of the test case.

Reply:
> 1. The final result of `ngx_tpl.lua` depends on the configuration of `yaml`, I don't understand what you mean, and all of your test cases are written like this:

Not exactly true, we have some `lua_shared_dict` defined directly in `ngx_tpl.lua` which does not appear in `config-default.yaml`, like `xds-config` `xds-config-version`, but maybe we can optimize it later.

Reply:
> nginx_config:
>   http:
>     lua_shared_dict:
>       worker-events: 10m
>       upstream-healthcheck: 10m
>       not-exist-shared-dict: 10m
> --- request
> GET /apisix/prometheus/metrics
> --- response_body_unlike
> qr/.*apisix_shared_dict_capacity_bytes{name="not-exist-shared-dict"} \d+(?:.|\n)*
> apisix_shared_dict_free_space_bytes{name="not-exist-shared-dict"} \d+.*/

Got it, but I don't think it's necessary to test whether shared dict works or not.

Reply:
> > 1. The final result of `ngx_tpl.lua` depends on the configuration of `yaml`, I don't understand what you mean, and all of your test cases are written like this:
> 
> Not exactly true, we have some `lua_shared_dict` defined directly in `ngx_tpl.lua` which does not appear in `config-default.yaml`, like `xds-config` `xds-config-version`, but maybe we can optimize it later.

Got it, but I don't think it's necessary to test whether shared dict works or not.

Reply:
I think this is just another way of describing the correctness of the returned data

Reply:
`yaml_config` can determine whether `ngx.shared` exists. For custom `ngx.shared`, the configuration of `yaml_config` is a necessary condition, but it is not a sufficient condition. Here it can be verified that `yaml_config` is not sufficient condition.

Reply:
Also the lua_shared_dict here is just to control the size of shdict. Adding the configuration or not doesn't affect the lua_shared_dict  in `nginx.conf`.

Reply:
> > Here it can be verified that yaml_config is not sufficient condition.
> 
> We already have some lua_shared_dict defined in the yaml by default. Is it necessary to override the default yaml to verify it?

I tested and found that APISIX will use `config-default.yaml` as the configuration when running the test case, my understanding was wrong.

Reply:
I tested and found that APISIX will use `config-default.yaml` as the configuration when running the test case, my understanding was wrong.